src/collector/top_collector.rs

```diff
@@ -86,32 +86,48 @@ where T: PartialOrd + Clone
 
     pub fn merge_fruits(
         &self,
-        children: Vec<Vec<(T, DocAddress)>>,
+        mut children: Vec<Vec<(T, DocAddress)>>,
     ) -> crate::Result<Vec<(T, DocAddress)>> {
         if self.limit == 0 {
             return Ok(Vec::new());
         }
-        let mut top_collector = BinaryHeap::new();
+        if children.len() == 1 {
+            let mut child = children.pop().unwrap();
+            child.sort_by(|a, b| {
+                if a.0 == b.0 {
+                    a.1.cmp(&b.1)
+                } else {
+                    b.0.partial_cmp(&a.0).unwrap_or(std::cmp::Ordering::Equal)
```

Comment:
Isn't this redundant to the call of `PartialEq` above? Something like

```rust
match b.0.partial_cmp(&a.0) {
  Some(Ordering::Equal) => a.1.cmp(&b.1),
  Some(other) => other,
  None => Ordering::Equal,
}
```

Or is it important to definitely call `PartialEq` first and not relying on `PartialEq` and `PartialCmp` agreeing on equality?

Reply:
I don't think that's important. I replaced that part use the comparison in `ComparableDoc`